climada/engine/forecast.py

```diff
@@ -0,0 +1,897 @@
+        Returns
+        -------
+        float
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return self._impact[haz_ind].aai_agg
+
+    def haz_summary_str(self, run_datetime=None):
+        """ provide a summary string for the hazard part of the forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        str
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        return (self.hazard[haz_ind].tag.haz_type +
+                '_' +
+                self.haz_model +
+                '_run' +
+                run_datetime.strftime('%Y%m%d%H') +
+                '_event' +
+                self.event_date.strftime('%Y%m%d')
+                )
+
+    def summary_str(self, run_datetime=None):
+        """ provide a summary string for the impact forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        str
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        return (self.haz_summary_str(run_datetime) +
+                '_' +
+                self.exposure_name
+                )
+
+    def lead_time(self, run_datetime=None):
+        """ provide the lead time for the impact forecast
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            select the used hazard by the run_datetime.
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        return self.event_date - run_datetime
+
+    def calc(self, force_reassign=False, check_plot=False):
+        """ calculate the impact using
+        exposure, hazard, and vulnerabilty.
+
+        Parameters
+        ----------
+        force_reassign: bool, optional
+            Reassign centroids.
+        check_plot: bool, optional
+            Show plot of expected annual impact if True.
+        """
+        # generate folders
+        if not Path(FORECAST_PLOT_DIR).exists():
+            Path(FORECAST_PLOT_DIR).mkdir(parents=True)
+
+        # force reassign
+        if force_reassign:
+            self.exposure.assign_centroids(self.hazard)
+        # calc impact
+        for ind_i, haz_i in enumerate(self.hazard):
+            self._impact[ind_i].calc(self.exposure,
+                         self.vulnerability,
+                         haz_i,
+                         save_mat=True)
+            if check_plot:
+                self._impact[ind_i].plot_hexbin_eai_exposure()
+
+    def plot_imp_map(self,
```

Comment:
For all the plots please do:

1) return the axes (and not the fig)
2) add a keyword argument `figsize` to the function to set the figsize (this is very hard to change afterwards).

Reply:
both are implemented